python/ray/tests/test_runtime_env_conda_and_pip_2.py

```diff
@@ -33,37 +37,36 @@ def test_install_failure_logging(
         if using_ray_client:
             with pytest.raises(ConnectionAbortedError) as excinfo:
                 ray.init(address, runtime_env=bad_envs["init"])
-            assert bad_packages["init"] in str(excinfo.value)
+                assert bad_packages["init"] in str(excinfo.value)
         else:
             ray.init(address, runtime_env=bad_envs["init"])
-            wait_for_condition(
-                lambda: bad_packages["init"] in capsys.readouterr().out, timeout=30
-            )
+
+            @ray.remote
+            def g():
+                pass
+
+            with pytest.raises(RuntimeEnvSetupError, match=bad_packages["init"]):
+                ray.get(g.remote())
         return
 
     ray.init(address)
 
     @ray.remote(runtime_env=bad_envs["actor"])
     class A:
-        pass
+        def f(self):
+            pass
 
     a = A.remote()  # noqa
-
-    wait_for_condition(
-        lambda: bad_packages["actor"] in capsys.readouterr().out, timeout=30
-    )
+    with pytest.raises(RuntimeEnvSetupError, match=bad_packages["actor"]):
+        ray.get(a.f.remote())
```

Comment:
What's the reason for adding f here?  Does it no longer error when you just create the actor, with A.remote()?

Reply:
We stopped printing logs at all, and it will be only transmitted from the exception.

So testing with capsys makes no sense anymore. Also, creating actors don't raise an exception by Ray's error handling model.

Reply:
Makes sense, thanks!  Is the best reference for the error handling model just the Ray whitepaper?

Reply:
I think it is pretty simple. All of the exceptions are raised when you ray.get on tasks (either task or actor tasks)